### PR TITLE
Use lld linker in CMake CI

### PR DESCRIPTION
**Pull request by GMNGeoffrey**
Fast is better than slow. The actual motivation for this came from
working on https://github.com/google/iree/pull/7126 and noticing that
the link was really slow with gcc. The link is still really slow with
gcc, but I think this makes it better across the board.

Some ancillary cleanup was needed for this:
- pinning the `keras-nightly` pip package version, otherwise rebuilding
the docker image picked up the latest, even with a pinned tf-nightly.
- making the check for unsupported linker options more robust by also
testing the C compiler.

### CODE REVIEW

File: build_tools/cmake/iree_setup_toolchain.cmake [hunk 1]
```diff
@@ -20,13 +20,25 @@ endif()
 if(IREE_USE_LINKER)
   set(IREE_LINKER_FLAG "-fuse-ld=${IREE_USE_LINKER}")
 
+  iree_append("${IREE_LINKER_FLAG}"
```
Comment (ScottTodd) [inline build_tools/cmake/iree_setup_toolchain.cmake:23]: change to `IREE_LINKER_FLAGS` since there are multiple flags in the variable now?

**Review by ScottTodd — APPROVED** [review]

File: build_tools/cmake/iree_setup_toolchain.cmake (hunk 1 above)
Comment (GMNGeoffrey) [reply build_tools/cmake/iree_setup_toolchain.cmake:23]: No there's still only one flag. This is using the helper function above which appends the first argument to all the trailing arguments

Comment (ScottTodd) [reply build_tools/cmake/iree_setup_toolchain.cmake:23]: Oh, that name could be more clear then :P 
`iree_append_to_lists` or something, idk

your change is fine as-is

Comment (GMNGeoffrey) [reply build_tools/cmake/iree_setup_toolchain.cmake:23]: Yeah agreed :-) Given that the implementation is like 10 lines up I wasn't too stressed about it though

**Pull request merged**